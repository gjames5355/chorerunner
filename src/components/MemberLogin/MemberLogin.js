import React, { Component } from 'react';
import AuthApiService from '../../services/auth-api-service';
import UserContext from '../../contexts/UserContext';

export default class MemberLogin extends Component {
  static contextType = UserContext;

  state = {
    username: '',
    password: '',
    error: null,
  };

  onChangeHandle = e => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  };

  onLoginSuccess = () => {
    const { location, history } = this.props;
    const destination = (location.state || {}).from || '/member-dashboard';
    history.push(destination);
  };

  handleSubmit = e => {
    e.preventDefault();
    const { username, password } = e.target;

    this.setState({ error: null });

    AuthApiService.postMemberLogin({
      username: username.value,
      password: password.value,
    })
      .then(res => {
        username.value = '';
        password.value = '';
        this.context.processLogin(res.authToken, res.type);
        this.onLoginSuccess();
      })
      .catch(res => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error, username, password } = this.state;

    return (
      <div className='parent-login'>
        <h2>Member login</h2>
        <div className='demo'>
          <h3>Demo:</h3>
          <p>username: bartman</p>
          <p>password: bartman</p>
        </div>
        <form
          className='parent-form-container'
          id='member-login'
          onSubmit={this.handleSubmit}
        >
          <div>
            <label htmlFor='username'>Username</label>
            <input
              name='username'
              type='text'
              onChange={this.onChangeHandle}
              value={username}
              required
            ></input>
          </div>
          <div>
            <label htmlFor='password'>Password</label>
            <input
              name='password'
              type='password'
              onChange={this.onChangeHandle}
              value={password}
              required
            ></input>
          </div>
          <div role='alert'>{error && <p>{error}</p>}</div>
          <button type='submit'>login</button>
        </form>
      </div>
    );
  }
}
