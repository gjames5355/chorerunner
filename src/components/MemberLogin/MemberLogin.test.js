import React from 'react';
import ReactDOM from 'react-dom';
import MemberLogin from './MemberLogin';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('MemberLogin component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <MemoryRouter>
        <MemberLogin />
      </MemoryRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <MemberLogin />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<MemberLogin />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('reads the components State', () => {
    const wrapper = shallow(<MemberLogin />);
    expect(wrapper.state('username')).toEqual('');
    expect(wrapper.state('password')).toEqual('');
    expect(wrapper.state('error')).toEqual(null);
  });

  it('checks for input attribute length', () => {
    const wrapper = shallow(<MemberLogin />);
    expect(wrapper.find('input').length).toEqual(2);
  });

  it('submit event gets called', () => {
    const func = jest.fn();
    const wrapper = shallow(<MemberLogin />);
    wrapper.find('form').contains('submit', { preventDefault: () => {} });
    expect(wrapper).toEqual({});
  });
});
