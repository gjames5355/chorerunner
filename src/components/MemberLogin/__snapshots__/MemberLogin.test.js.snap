// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberLogin component testing renders UI as expected 1`] = `
<div
  className="parent-login"
>
  <h2>
    Member login
  </h2>
  <div
    className="demo"
  >
    <h3>
      Demo:
    </h3>
    <p>
      username: bartman
    </p>
    <p>
      password: bartman
    </p>
  </div>
  <form
    className="parent-form-container"
    id="member-login"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        name="username"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      role="alert"
    />
    <button
      type="submit"
    >
      login
    </button>
  </form>
</div>
`;

exports[`MemberLogin component testing renders empty given no tabs 1`] = `
<div
  className="parent-login"
>
  <h2>
    Member login
  </h2>
  <div
    className="demo"
  >
    <h3>
      Demo:
    </h3>
    <p>
      username: bartman
    </p>
    <p>
      password: bartman
    </p>
  </div>
  <form
    className="parent-form-container"
    id="member-login"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        name="username"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      role="alert"
    />
    <button
      type="submit"
    >
      login
    </button>
  </form>
</div>
`;
